src/universe/visualization/img_tools.py

```diff
@@ -14,3 +18,313 @@ def rotate(img: np.ndarray, angle: float) -> np.ndarray:
     rot_mat = cv.getRotationMatrix2D((cols // 2, rows // 2), angle, 1)
 
     return cv.warpAffine(img, rot_mat, (cols, rows))
+
+class _Layer:
+    """
+    A layer that can be drawn on an image.
+    """
+
+    def __init__(self, color: Tuple[int, int, int], opacity: int, name: str = None):
+        """
+        Creates a new layer with the given color, opacity and name.
+
+        :param color: color of the layer (RGB)
+        :param opacity: opacity of the layer (0-100)
+        :param name: name of the layer
+        """
+        self._opacity = None
+        self.set_opacity(opacity)
+        self.color = color
+        self.name = name
+
+    def set_opacity(self, opacity: int):
+        """
+        Sets the opacity of the layer.
+
+        :param opacity: opacity of the layer (0-100)
+        """
+        assert 0 <= opacity <= 100, 'Alpha must be between 0 and 100'
+        self._opacity = opacity
+
+    def draw_on(self, image: cv.Mat) -> cv.Mat: ...
```

Comment:
If the draw_on method is meant to be overridden by subclasses, then it should be an abstract method:
```suggestion
    @abstractmethod
    def draw_on(self, image: cv.Mat) -> cv.Mat: 
        pass
```

Also, the _Layer class should extend the ABC class from abc.